### ISSUE TITLE
runtime attribute override behavior depends upon variable name

### ISSUE THREAD

**Issue by MatthewMah** [broadinstitute/cromwell#2068]
I have configured a SLURM backend for Cromwell and have encountered unusual behavior while trying to configure memory as a runtime attribute. 

I define a runtime attribute Int with default value in my Cromwell configuration file and attempt to override this in my task WDL. Whether the override succeeds seems to depend on the variable name used! This is very confusing behavior; I expect to either receive a message that a variable name is not allowed, or the override should succeed. 

Fails: "memory_mb"
Succeeds: "requested_memory_per_core"

I am verifying whether the override succeeds by checking the Cromwell output [task]/execution/script.submit. 

```
    SLURM {
      actor-factory = "cromwell.backend.impl.sfs.config.ConfigBackendLifecycleActorFactory"
      config {
       runtime-attributes = """
        Int runtime_minutes = 600
        Int bsub_cpu = 1
        Int memory_mb = 1000
        String queue
        """
      
        submit = """
			sbatch -J ${job_name} -D ${cwd} -o ${out} -e ${err} -t ${runtime_minutes} -p ${queue} \
			${"-n " + bsub_cpu} \
			--mem-per-cpu=${memory_mb} \
			--wrap "/bin/bash ${script}"
		"""
        kill = "scancel ${job_id}"
        check-alive = "squeue -j ${job_id}"
        job-id-regex = "Submitted batch job (\\d+).*"
      }
    }
```

```
workflow tutorialWorkflow{
	call task_A { input : in="testing" }
	call task_B { input : in=task_A.out }
	call task_C { input : in=task_A.out }
}

task task_A{
	String in

	command{
		echo 'This is task A ${in}.'
	}	
	output{
		String out='This is task A ${in}'
	}
	runtime{
		bsub_cpu: 1
		runtime_minutes: 10
		memory_mb: 100
		queue: "short"
	}
}

task task_B{
	String in
	command{
		echo 'This is task B ${in}.'
	}
	runtime{
		bsub_cpu: 2
		runtime_minutes: 15
		memory_mb: 110
		queue: "short"
	}
}

task task_C{
	String in
	command{
		echo 'This is task C ${in}.'
	}
	runtime{
		runtime_minutes: 25
		memory_mb: 210
		queue: "short"
	}
}
```

**Comment by kshakir** [comment]
In the [README](https://github.com/broadinstitute/cromwell/blob/25/README.md#sun-gridengine-backend), it's buried and could be clearer, but it says:

> There are two special runtime attribute configurations, cpu, and memory_\<unit\>.
> …
> When the runtime attribute configuration Int memory_\<unit\> or Float memory_\<unit\> is specified, it is provided to submit by the runtime attribute in WDL memory.

(slightly better formatting in the README)

There's more "what" in the README, but the "why" extends from the fact that JES, TES, and other backends all use a common `memory` runtime attribute. Using this common runtime attribute name, `memory`, increases the chance that a WDL will be runnable on a different backends.

In a WDL run on standard backends, `memory` is specified as a WDL string, parsed by the backends into a [`MemorySize`](https://github.com/broadinstitute/cromwell/blob/25/backend/src/main/scala/cromwell/backend/MemorySize.scala#L39). However, the Config backend used for SLURM, SGE, Local, etc., needs to covert the `MemorySize` back into a string, for embedding into the custom `submit` string.

The `_<unit>` in `memory_<unit>` is how the `MemoryUnit` gets converted into a string. Say someone defines a WDL originally intended to run on the JES backend, containing a task with `memory: "2 GB"`. If instead, this same WDL will be run on a Config backend, and the config specifies `Int memory_mb`, the string value of `memory_mb` passed into `submit` will expand to `2000`.

Let us know if you have more questions and/or suggestions, or if this resolves this particular issue for now?

**TL;DR memory_\<unit\> is one of the reserved runtime-attribute names, meant to make WDLs more portable.**

**Comment by MatthewMah** [comment]
Yes, this explains my issue. I suggest including a description of runtime parameters in the user guide: https://software.broadinstitute.org/wdl/userguide/

**Comment by katevoss** [comment]
@MatthewMah thanks for your feedback, I'm in the process of updating the Cromwell docs and information about the runtime parameters will be a part of that.
